google/cloud/storage/client.h

```diff
@@ -251,25 +249,48 @@ class Client {
       : Client(ClientOptions(std::move(credentials)),
                std::forward<Policies>(policies)...) {}
 
+  /// Create a Client using ClientOptions::CreateDefaultClientOptions().
+  static StatusOr<Client> CreateDefaultClient();
+
   /// Builds a client and maybe override the retry, idempotency, and/or backoff
   /// policies.
+  /// @deprecated This was intended only for test code, applications should not
+  /// use it.
   template <typename... Policies>
+#if !defined(_MSC_VER) || _MSC_VER >= 1920
+  GOOGLE_CLOUD_CPP_DEPRECATED(
+      "applications should not need this."
+      " Please use the constructors from ClientOptions instead."
+      " For mocking, please use testing::ClientFromMock() instead."
+      " Please file a bug at https://github.com/googleapis/google-cloud-cpp"
+      " if you have a use-case not covered by these.")
+#endif  // _MSC_VER
+  // NOLINTNEXTLINE(performance-unnecessary-value-param)
   explicit Client(std::shared_ptr<internal::RawClient> client,
                   Policies&&... policies)
-      : raw_client_(
-            Decorate(std::move(client), std::forward<Policies>(policies)...)) {}
+      : Client(InternalOnly{}, std::move(client),
+               std::forward<Policies>(policies)...) {
```

Comment:
@coryan ... does this ctor exist?  It seems not, but maybe I'm missing something.

Reply:
I agree. Let me prepare a PR to fix this.

Reply:
Sent #6439 let's continue the discussion there.